rust.c

```diff
@@ -0,0 +1,984 @@
+/*
+*
+*   This source code is released for free distribution under the terms of the
+*   GNU General Public License.
+*
+*   This module contains functions for generating tags for Rust files.
+*/
+
+/*
+*   INCLUDE FILES
+*/
+#include "general.h"	/* must always come first */
+#include "main.h"
+
+#include <string.h>
+
+#include "keyword.h"
+#include "parse.h"
+#include "entry.h"
+#include "options.h"
+#include "read.h"
+#include "vstring.h"
+
+/*
+*   MACROS
+*/
+#define MAX_STRING_LENGTH 256
+
+/*
+*   DATA DECLARATIONS
+*/
+
+typedef enum {
+	K_MOD,
+	K_STRUCT,
+	K_TRAIT,
+	K_IMPL,
+	K_FN,
+	K_ENUM,
+	K_TYPE,
+	K_STATIC,
+	K_MACRO,
+	K_FIELD,
+	K_VARIANT,
+	K_METHOD,
+	K_NONE
+} RustKind;
+
+static kindOption rustKinds[] = {
+	{TRUE, 'n', "namespace", "module"},
```

Comment:
> Whatever makes it easier I suppose...

For Geany compatibility keeping the current values is a bit easier, but if they don't change over time (and they shouldn't) it's not very much of a concern as it wouldn't affect a diff.

>  if I recall correctly these strings are not user-visible anyway?

In Geany they aren't really, but in CTags they end up in the tag file as the tag kind.  How it is used will depend on the application I guess, but I guess they might end up being the label used to represent their kind, or they might get mapped back to known kinds (kind of like in Geany, and as I plan to do in Geany at some point so that we don't have to alter them at the parser level).
So… I guess the best is to have meaningful names, but it's probably not so important either.